crates/hir_ty/src/lib.rs

```diff
@@ -1535,16 +1548,18 @@ mod tests {
     fn infer_function_param() {
         check_in_root_file(
             r#"
-                fn aaa(x: int, y: string) {
-                    let a = x
-                    let b = y
+                fn aaa(x: int, y: string, z: mut bool) {
+                    let a = x;
+                    let b = y;
+                    let c = z;
                 }
             "#,
             expect![[r#"
                 //- /main.nail
-                fn aaa(x: int, y: string) -> () {
+                fn aaa(x: int, y: string, z: mut bool) -> () {
                     let a = param:x; //: int
                     let b = param:y; //: string
+                    let c = param:z; //: bool
```

Comment:
**Scalability Issue**: Introduction of mutable state in function parameters increases complexity in concurrent environments. Mutable state can lead to data races if not handled properly in a concurrent setup. <br> **Fix**: Consider using immutable data structures or ensure proper synchronization mechanisms are in place if mutability is necessary. <br> **Code Suggestion**: 
 ```
 -                fn aaa(x: int, y: string, z: mut bool) {
 +                fn aaa(x: int, y: string, z: bool) {
 -                    let c = param:z; //: bool
 +                    let c = param:z.clone(); //: bool
 ```